composite/synth-index/src/adapter.ts

```diff
@@ -0,0 +1,54 @@
+import { Validator, Requester } from '@chainlink/external-adapter'
+import { AdapterResponse, Execute, AdapterRequest } from '@chainlink/types'
+import {
+  types,
+  makeExecute as tokenAllocationMakeExecute,
+} from '@chainlink/token-allocation-adapter'
+import snx from 'synthetix'
+import { makeConfig, Config } from './config'
+
+const customParams = {
+  asset: ['asset', 'from'],
+  network: false,
+}
+
+type Synth = {
+  name: string
+  asset: string
+  index: any[]
+  inverted: Record<string, any>
+}
+
+const getAllocations = (synth: Synth): types.TokenAllocations => {
+  return synth.index.map((index) => ({
+    symbol: index.asset,
+    balance: index.units,
+    decimals: 0,
```

Comment:
The units are already formatted taking in account the token decimals

Reply:
I think the `toValidAllocations` fn in `token-allocation` is going to give you problems with `decimals: 0`:

```javascript
decimals: t.decimals || DEFAULT_TOKEN_DECIMALS,
```

It is going to set decimals as 18, the default.

Reply:
Ok, I see now this is addressed later.